### DISCUSSION TITLE
Optional[Department] vs Optional[Union[Department, Dict]]

### DISCUSSION THREAD

**Discussion by Erhannis** [collerek/ormar#100] [Q&A] [answered]
Question: the docs sometimes have e.g.
```
department: Optional[Union[Department, Dict]] = ormar.ForeignKey(Department)
```
and sometimes
```
department: Optional[Department] = ormar.ForeignKey(Department)
```

Is there a reason for the difference?

Also, is this the best place to ask questions, or is there a forum or chat or something?  Or should I use StackOverflow?

**Comment by collerek** [comment, marked as answer]
Ok, so I enabled the github discussions as a place for asking questions and discussing changes etc.

The annotations are totally optional and everything will work as expected without them.
The relations fields are actually quite complex and hard to annotate as they behave like this:

You always get in return (for FK field): `ormar.Model` instance or None

**but** you can initialize a related model in several ways:

* with `ormar.Model` instance (pass Department instance in example above)
* with dictionary containing keys matching model names
* with primary key field value (so depends on the type of the primary key column: string, int, uuid etc.) < - so basically Any
* None

So when you put `department: Optional[Department]` and later try to initialize it with a dict your ide/ static type checking tool will misbehave/complain :) So the annotation usually matches what I do in code, as most of the examples in docs are taken from tests. 

Probably I should provide some kind of custom type with overloading to annotate relations but as you can see above it is not trivial.

**Reply by Erhannis** [reply]
Neat, I've not seen Github Discussions before, but it looks useful.  I may be peppering you with small clarification questions; hope that's ok....

Oh, ok; makes sense, thanks.

And oh, interesting - so, suppose I have an object that has an FK to another object with e.g. a uuid ID, could I send to a POST endpoint a json like `{"data":"blah","foreign_object":"a51692-uuid-whatever-1423415}`, and it'd end up with the relevant foreign_object?  ...Upon testing, looks like so, yes.  Neat!
